controllers/operatorconfigmap_controller.go

```diff
@@ -0,0 +1,454 @@
+			csi.GetRBDDaemonSet(c.OperatorNamespace).DeepCopyInto(c.rbdDaemonSet)
+			if err := c.own(c.rbdDaemonSet); err != nil {
+				return err
+			}
+			return nil
+		})
+		if err != nil {
+			c.log.Error(err, "failed to create/update rbd daemonset")
+			return ctrl.Result{}, err
+		}
+
+		// Need to handle deletion of the csiDriver object, we cannot set
+		// ownerReference on it as its cluster scoped resource
+		cephfsCSIDriver := templates.CephFSCSIDriver.DeepCopy()
+		cephfsCSIDriver.ObjectMeta.Name = csi.GetCephFSDriverName()
+		if err := csi.CreateCSIDriver(c.ctx, c.Client, cephfsCSIDriver); err != nil {
+			c.log.Error(err, "unable to create cephfs CSIDriver")
+			return ctrl.Result{}, err
+		}
+
+		rbdCSIDriver := templates.RbdCSIDriver.DeepCopy()
+		rbdCSIDriver.ObjectMeta.Name = csi.GetRBDDriverName()
+		if err := csi.CreateCSIDriver(c.ctx, c.Client, rbdCSIDriver); err != nil {
+			c.log.Error(err, "unable to create rbd CSIDriver")
+			return ctrl.Result{}, err
+		}
+
+		prometheusRule := &monitoringv1.PrometheusRule{}
+		if err := k8sYAML.NewYAMLOrJSONDecoder(bytes.NewBufferString(string(pvcPrometheusRules)), 1000).Decode(prometheusRule); err != nil {
+			c.log.Error(err, "Unable to retrieve prometheus rules.", "prometheusRule", klog.KRef(prometheusRule.Namespace, prometheusRule.Name))
+			return ctrl.Result{}, err
+		}
+
+		prometheusRule.SetNamespace(c.OperatorNamespace)
+
+		err = c.createOrUpdate(prometheusRule, func() error {
+			applyLabels(c.operatorConfigMap.Data["OCS_METRICS_LABELS"], &prometheusRule.ObjectMeta)
+			return c.own(prometheusRule)
+		})
+		if err != nil {
+			c.log.Error(err, "failed to create/update prometheus rules")
+			return ctrl.Result{}, err
+		}
+
+		c.log.Info("prometheus rules deployed", "prometheusRule", klog.KRef(prometheusRule.Namespace, prometheusRule.Name))
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (c *OperatorConfigMapReconciler) deletionPhase(req *ctrl.Request) error {
+	csiAddonsNode := &csiaddonsv1alpha1.CSIAddonsNode{}
+	err := c.Client.Get(c.ctx, req.NamespacedName, csiAddonsNode)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// Request object not found, could have been deleted before reconcile request.
+			c.log.Info("CSIAddonsNode resource not found")
+			return nil
+		}
+		return err
+	}
+	csiAddonsNode.Finalizers = nil
+	if err := c.Client.Update(c.ctx, csiAddonsNode); err != nil {
+		c.log.Error(err, "Failed to remove finalizer from csiAddonsNode")
+		return err
+	}
+	return nil
+}
+func (c *OperatorConfigMapReconciler) createOrUpdate(obj client.Object, f controllerutil.MutateFn) error {
+	result, err := controllerutil.CreateOrUpdate(c.ctx, c.Client, obj, f)
+	if err != nil {
+		return err
+	}
+	c.log.Info("successfully created or updated", "operation", result, "name", obj.GetName())
+	return nil
+}
+
+func (c *OperatorConfigMapReconciler) own(obj client.Object) error {
+	return controllerutil.SetControllerReference(c.OperatorDeployment, obj, c.Client.Scheme())
```

Comment:
I think this is wrong. It `r.Own` should set the config map as the owning resource. It might be the exact issue we see with some resources no being deleted when the config map is deleted.

Reply:
Nice catch! yes certainly needs a fix, done.

Reply:
@bernerhat Please don't resolve any conversation until you get an ack that the reviewer who issued the comment is ok with the fix. You have 2 options:
1. if the reviewer is a maintainer and can resolve the comments themselves then let them do it.
2. If they are not, wait for an ack on the comment thread and only then resolve.